File: src/main/java/com/vinyl/model/Band.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Band {
	private String bandAlias;
	private Boolean isBandActive;
	private String bandCountryCode;
	private Date startYear;
	private Date endYear;
	private List<String> currentArtistAliases;
	private List<String> previousArtistAliases;
	private List<String> trackCatalogNums;
	private List<String> featuringTracksCatalogNums;
}

File: src/main/resources/frontend/src/schemas/definitions/band.js
export default {
  props: {
    bandAlias: 'string',
    isBandActive: 'boolean',
    bandCountryCode: 'country',
    startYear: {
      type: 'date',
      typeConstraint: {
        to: []
      }
    },
    endYear: {
      type: 'date',
      isNullable: true,
      typeConstraint: {
        to: []
      }
    },
    currentArtistAliases: {
      type: 'artist',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    previousArtistAliases: {
      type: 'artist',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    trackCatalogNums: {
      type: 'track',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    featuringTracksCatalogNums: {
      type: 'track',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    participations: {
      type: 'participation',
      isSchema: true,
      isArray: true,
      filterOnly: true,
      joins: ['artist2band']
    }
  },
  key: 'bandAlias',
  display: (item) => item.bandAlias
}